e2e/cypress/integration/api/iam/14_projects_delete_api.spec.ts

```diff
@@ -0,0 +1,146 @@
+import { describeIfIAMV2p1 } from '../../../support/constants';
+import { uuidv4 } from '../../../support/helpers';
+
+describeIfIAMV2p1('Project delete', () => {
+  const cypressPrefix = 'test-project-delete';
+
+  const project =  {
+    id: `${cypressPrefix}-project-delete-applied-rule${Cypress.moment().format('MMDDYYhhmm')}`,
+    name: 'project delete'
+  };
+
+  const rule = {
+    id: 'rule-applied',
+    name: 'rule',
+    type: 'NODE',
+    project_id: project.id,
+    conditions: [
+      {
+        attribute: 'CHEF_SERVER',
+        operator: 'EQUALS',
+        values: ['example.org']
+      }
+    ]
+  };
+
+  before(() => {
+    cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']);
+  });
+
+  after(() => cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']));
+
+  it('deleting a project', () => {
+    // Create a project
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: '/apis/iam/v2beta/projects',
+      body: project
+    });
+
+    // Add a rule to the project
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'POST',
+      url: `/apis/iam/v2beta/projects/${project.id}/rules`,
+      body: rule
+    });
+
+    // Try to delete the project with a stagged rule
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'DELETE',
+      url: `/apis/iam/v2beta/projects/${project.id}`,
+      failOnStatusCode: false
+    }).then((deleteResp) => {
+      expect(deleteResp.status,
+        'Did not fail deleting a project with a stagged rule').to.be.oneOf([400]);
+    });
+
+    // Ensure the project was not deleted
+    cy.request({
+      headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+      method: 'GET',
+      url: '/apis/iam/v2beta/projects'
+    }).then((resp: Cypress.ObjectLike) => {
+      expect(resp.body.projects && resp.body.projects.length > 0 &&
```

Comment:
These read a bit easier as separate lines. Cypress has a [helpful list of all the assertions it allows](https://docs.cypress.io/guides/references/assertions.html#BDD-Assertions)

Reply:
We can not use a lot of those assertions because the linter does not like them. For example `expect(myVar).to.exist` and `expect(true).to.be.true`. I get the error "expected an assignment or function call".

Reply:
After considering the various operators, I believe what you have is optimal. 
(Here's one source that concurs: https://stackoverflow.com/a/34445397)
However, since you reuse this check several times, consider extracting it as a helper function:
```
      expect(responseContainsProject(resp.body.projects, project.id)).to.be.equal(true);
```